awx/main/tasks/jobs.py

```diff
@@ -1056,18 +1059,18 @@ def pre_run_hook(self, job, private_data_dir):
             error = _('Job could not start because no Execution Environment could be found.')
             self.update_model(job.pk, status='error', job_explanation=error)
             raise RuntimeError(error)
-        elif job.project.status in ('error', 'failed'):
-            msg = _('The project revision for this job template is unknown due to a failed update.')
-            job = self.update_model(job.pk, status='failed', job_explanation=msg)
-            raise RuntimeError(msg)
 
         if job.inventory.kind == 'smart':
             # cache smart inventory memberships so that the host_filter query is not
             # ran inside of the event saving code
             update_smart_memberships_for_inventory(job.inventory)
 
     def build_project_dir(self, job, private_data_dir):
-        self.sync_and_copy(job.project, private_data_dir, scm_branch=job.scm_branch)
+        try:
+            self.sync_and_copy(job.project, private_data_dir, scm_branch=job.scm_branch)
+        except RuntimeError as e:
+            self.update_model(job.pk, status='failed', job_explanation=str(e))
```

Comment:
Super nit pick comment, but I'd like to put the `update_model` call for job_explanation into the main `sync_and_copy` method for consistency between the two types.

Reply:
done